VProperty::removeParameter(wchar_t* paramName) can only
be used for parameters which are guaranteed to be unique.
This not the case e.g. TYPE, so another version which
removes parameter by index is needed.

#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- native/src/c++/common/vocl/VProperty.cpp~vocl-removeParameter
+++ native/src/c++/common/vocl/VProperty.cpp
@@ -105,6 +105,11 @@
         }
     }
 }
+
+void VProperty::removeParameter(int index) {
+    parameters->remove(index);
+}
+   
 bool VProperty::containsParameter(wchar_t* paramName) {
     
     if (parameters != NULL) {
--- native/src/include/common/vocl/VProperty.h~vocl-removeParameter
+++ native/src/include/common/vocl/VProperty.h
@@ -46,6 +46,7 @@
     wchar_t* getValue(wchar_t* buf = NULL, int size = -1);
     void addParameter(const wchar_t* paramName, const wchar_t* paramValue);
     void removeParameter(wchar_t* paramName);
+    void removeParameter(int index);
     bool containsParameter(wchar_t* paramName);
     // Warning: the name does not always uniquely identify
     // the parameter, some of them may occur multiple times.
